src/renderer/store.js

```diff
@@ -0,0 +1,90 @@
+import { create } from 'zustand'
+import { persist, createJSONStorage } from 'zustand/middleware'
+import store from '../persist-store'
+
+/**
+ * @type {import('zustand/middleware').StateStorage}
+ */
+const storage = {
+  /**
+   * @param {string}  key - A string key to reference the stored item by.
+   * @returns {Promise<string | null>}
+   */
+  getItem: async key => {
+    return (await store.get(key)) || null
```

Comment:
there is a typing issue here. Typescript is expecting `string` or `null`. `store.get` return `unknown`. You will have to cast a type here